jcl/src/java.base/share/classes/jdk/crac/Core.java

```diff
@@ -0,0 +1,109 @@
+/*[INCLUDE-IF CRAC_SUPPORT]*/
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2024
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+package jdk.crac;
+
+import java.nio.file.Paths;
+
+import openj9.internal.criu.InternalCRIUSupport;
+import openj9.internal.criu.JVMCheckpointException;
+import openj9.internal.criu.JVMRestoreException;
+
+/**
+ * The CRIU Support API.
+ */
+public class Core {
+	private static CRIUSupportContext<?> globalContext = new CRIUSupportContext<>();
+
+	/**
+	 * Get a global context.
+	 *
+	 * @return a Context
+	 */
+	public static Context<Resource> getGlobalContext() {
+		return (Context<Resource>)globalContext;
+	}
+
+	/**
+	 * Invoke InternalCRIUSupport with default settings.
+	 *
+	 * @throws CheckpointException if this failed with a CheckpointException
+	 * @throws RestoreException    if this failed with a RestoreException
+	 */
+	public static void checkpointRestore() throws CheckpointException, RestoreException {
+		try {
+			if (InternalCRIUSupport.isCRaCSupportEnabled() && InternalCRIUSupport.isNativeLoaded()) {
+				globalContext.checkpointJVM();
+			}
```

Comment:
we should throw a checkpoint exception if this fails for any reason, including crac. not being enabled

Reply:
Throw `UnsupportedOperationException` if `InternalCRIUSupport.isCRaCSupportEnabled()` is not `true`.

Also noticed that `InternalCRIUSupport.init()` wasn't invoked.
https://github.com/eclipse-openj9/openj9/blob/78e3e30edc2f21748b1f1a45a7ae7f33179da9c7/jcl/src/java.base/share/classes/openj9/internal/criu/InternalCRIUSupport.java#L227-L231

Is `libj9criu29` still relevant?

Reply:
Opened 
* https://github.com/eclipse-openj9/openj9/pull/18745